zerver/lib/actions.py

```diff
@@ -2733,9 +2733,26 @@ def check_message(
             type=Recipient.STREAM,
         )
 
+        sub = None
+        sub_user = None
+        if not sender.is_bot:
+            sub_user = sender
+        elif sender.bot_owner is not None:
+            sub_user = sender.bot_owner
+
+        if sub_user is not None:
+            try:
+                sub = Subscription.objects.get(
+                    user_profile=sub_user, recipient=recipient, active=True
+                )
+            except Subscription.DoesNotExist:
+                sub = None
+
+        # This will raise JsonableError if there are problems.
```

Comment:
Hmm, what's the thinking around this logic being here rather than in `access_stream_for_send_message`?  It doesn't look like we end up passing the `sub` object.